metpy/calc/indices.py

```diff
@@ -202,3 +202,100 @@ def bulk_shear(pressure, u, v, heights=None, bottom=None, depth=None):
     v_shr = v_layer[-1] - v_layer[0]
 
     return u_shr, v_shr
+
+
+@exporter.export
+def supercell_composite(mucape, esrh, ebwd):
+    r"""Calculate the supercell composite parameter.
+
+    The supercell composite parameter is designed to identify
+    environments favorable for the development of supercells,
+    and is calculated using the formula developed by
+    [Thompson2004]:
+
+    SCP = (mucape / 1000 J/kg) * (esrh / 50 m^2/s^2) * (ebwd / 20 m/s)
+
+    The ebwd term is set to zero below 10 m/s and
+    capped at 1 when ebwd exceeds 20 m/s.
+
+    Parameters
+    ----------
+    mucape : `pint.Quantity`
+        Most-unstable CAPE
+    esrh : `pint.Quantity`
+        Effective-layer storm-relative helicity
+    ebwd : `pint.Quantity`
+        Effective bulk shear
+
+    Returns
+    -------
+    array-like
+        supercell composite
+
+    """
+    ebwd = ebwd.to('m/s')
+    ebwd = ebwd.magnitude
+    inds = np.where((ebwd <= 20.) & (ebwd >= 10.))
+    inds1 = np.where(ebwd < 10.)
+    inds2 = np.where(ebwd > 20.)
+    ebwd[inds] = ebwd[inds] / 20.
+    ebwd[inds1] = 0.
+    ebwd[inds2] = 1.
+    sup_comp = (mucape.magnitude / 1000.) * (esrh.magnitude / 50.) * ebwd
```

Comment:
How about:
```python
return ((mucape / 1000 * units('J/kg')) *
             (effective_storm_helicity / 50 * units('m^2/s^2')) * effective_bulk_shear).to('dimensionless')
```
At a minimum, the old version would break when mucape or effective_storm_helicity were in different units.